Use consistent timezone for parsing

pkarman opened the pull request:
**Why**: Without explicit call to `.zone` the parse() method
can do unexpected things.

jessieay commented:
would it be possible to write a test that shows this unexpected behavior? I am curious on what that case would be. Would be good to prevent future regressions

monfresh commented:
I've been meaning to do this. Could you also please look for other instances of `Time.current` or `Time.now` and replace them with `Time.zone.now` so that we are consistent?

jessieay commented:
@monfresh I believe `Time.current` is an alias for `Time.zone.now` http://api.rubyonrails.org/classes/Time.html#method-c-current

pkarman commented:
`Time.current` is kind of an alias, but can act differently depending on config.

@jessieay I believe you pointed out somewhere that we are using `Eastern Time (US & Canada)` instead of `UTC` in `config/application.rb`. I have no idea why. I would prefer to use UTC everywhere always forever and ever amen. My suspicion is that the original code base was deployed for users primarily on the East cost and we just never changed it.

This came to my attention because my local version of rubocop warned about `DateTime.parse` with no explicit zone set. I agree that a single way of accessing the current time and zone is important.

Can we open a new issue for that bigger fix?

jessieay commented:
I like all those ideas @pkarman 

I still think that in the context of this PR it would be good to have a test that fails without this change since I am not sure why this was not working before (ugh time!)

monfresh commented:
Yep, except it uses a conditional to determine whether to use Time.zone.now or Time.now. Since we have the zone configured, it will always return Time.zone.now, so might as well use that. Either way, I don't have strong feelings as long as we're consistent.

monfresh commented:
If I understand Peter's comment about Rubocop correctly, it sounds like perhaps Code Climate does not run the Rails-specific checks, whereas our `make lint` uses `rubocop -R`.

pkarman commented:
I added the `zonebie` gem which randomly assigns timezone each time rspec runs. If you use the original code with the test I've added, it will fail every time the timezone is ahead of ET.

The reason it would otherwise pass every time is that because a Time object serializes with its timezone explicit, the parsing is un-ambiguous. It's when the stringified Time does not have an explicit timezone that we might run into trouble. In practice that never happens, which is why I had to construct a fictional test case.

Mostly I just wanted to quiet my local rubocop warnings. :) I suspect @monfresh is correct and codeclimate is running with different options (or a different version).
